### PR TITLE
SFX-142: Update sayt-client dependencies

### PR DESCRIPTION
**Pull request by DanielMoniz-GBI**
Updated the dependencies in sayt-client. Note that this involved introducing at least one breaking change.

Additionally, certain dependencies were left as-is, and one dependency was no longer needed and therefore removed.
* @types/es6-shim was removed
* codacy-coverage was not updated because it will be removed in another PR.

### CODE REVIEW

File: src/core/utils.ts [hunk 1]
```diff
@@ -3,6 +3,6 @@ import rawJsonp = require('jsonp');
 
 export function jsonp(url: string, body: any): Promise<any> {
   return new Promise((resolve, reject) => {
-    rawJsonp(`${url}?${qs.stringify(body)}`, (err, data) => err ? reject(err) : resolve(data));
+    rawJsonp(`${url}?${qs.stringify(body)}`, {}, (err, data) => err ? reject(err) : resolve(data));
```
Comment (jmbtutor) [inline src/core/utils.ts]: If `jsonp` didn't update, why is this change necessary? It looks like [`options` is an optional object](https://github.com/webmodules/jsonp/blob/790398945dc391ad2f06b365193a040a3dc01977/index.js#L40) anyway.

File: package.json [hunk 2]
```diff
@@ -38,42 +38,41 @@
...
+    "typedoc": "^0.14.2",
+    "typescript": "^3.5.2",
     "typings": "^2.1.1",
-    "webpack": "^2.4.1"
+    "webpack": "^4.35.2"
   },
   "dependencies": {
-    "@types/es6-shim": "^0.31.33",
-    "@types/node": "^7.0.14",
+    "@types/node": "12.0.10",
```
Comment (jmbtutor) [inline package.json]: Is there a reason why we're pinning it to this specific version?

File: src/core/utils.ts (hunk 1 above)
Comment (DanielMoniz-GBI) [reply src/core/utils.ts]: `options` may be optional, but we're passing the next optional argument.

I'll look into whether or not `jsonp` updated. I was getting errors until I made this change.

File: package.json (hunk 2 above)
Comment (DanielMoniz-GBI) [reply package.json]: You have the eyes of a hawk and the ears of a fox. Fixed.
